Lib/test/test_remote_pdb.py

```diff
@@ -0,0 +1,676 @@
+            stdout, _ = process.communicate(timeout=5)
+
+            self.assertIn("Function returned: 42", stdout)
+            self.assertEqual(process.returncode, 0)
+
+    def test_keyboard_interrupt(self):
+        """Test that sending keyboard interrupt breaks into pdb."""
+        script = f"""
+import time
+import sys
+import pdb
+def bar():
+    frame = sys._getframe()  # Get the current frame
+    pdb._connect(
+        host='127.0.0.1',
+        port={self.port},
+        frame=frame,
+        commands="",
+        version=pdb._PdbServer.protocol_version(),
+    )
+    print("Connected to debugger")
+    iterations = 10
+    while iterations > 0:
+        print("Iteration", iterations)
+        time.sleep(1)
+        iterations -= 1
+    return 42
+
+if __name__ == "__main__":
+    print("Function returned:", bar())
+"""
+        self._create_script(script=script)
+        process, client_file = self._connect_and_get_client_file()
+
+        with process:
+
+            # Skip initial messages until we get to the prompt
+            self._read_until_prompt(client_file)
+
+            # Continue execution
+            self._send_command(client_file, "c")
+
+            # Send keyboard interrupt signal
+            self._send_command(client_file, json.dumps({"signal": "INT"}))
+            self._send_interrupt(process.pid)
+            messages = self._read_until_prompt(client_file)
+
+            # Verify we got the keyboard interrupt message
+            interrupt_msg = next(msg['message'] for msg in messages if 'message' in msg)
+            self.assertIn("bar()", interrupt_msg)
+
+            # Continue to end
+            self._send_command(client_file, "iterations = 0")
+            self._send_command(client_file, "c")
+            stdout, _ = process.communicate(timeout=5)
+            self.assertIn("Function returned: 42", stdout)
+            self.assertEqual(process.returncode, 0)
+
+    def test_handle_eof(self):
+        """Test that EOF signal properly exits the debugger."""
+        self._create_script()
+        process, client_file = self._connect_and_get_client_file()
+
+        with process:
+            # Skip initial messages until we get to the prompt
+            self._read_until_prompt(client_file)
+
+            # Send EOF signal to exit the debugger
+            client_file.write(json.dumps({"signal": "EOF"}).encode() + b"\n")
+            client_file.flush()
+
+            # The process should complete normally after receiving EOF
+            stdout, stderr = process.communicate(timeout=5)
+
+            # Verify process completed correctly
+            self.assertIn("Function returned: 42", stdout)
+            self.assertEqual(process.returncode, 0)
+            self.assertEqual(stderr, "")
+
+    def test_protocol_version(self):
+        """Test that incompatible protocol versions are properly detected."""
+        # Create a script using an incompatible protocol version
+        script = f"""
```

Comment:
Could you do `textwrap.dedent()` for all the scripts? So it's visually indented properly.

Reply:
Sure

Reply:
@gaogaotiantian done